src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -83,181 +64,106 @@ describe('DOMPropertyOperations', () => {
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-    });
-
     it('should remove attributes for normal properties', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'title', 'foo');
-      expect(stubNode.getAttribute('title')).toBe('foo');
-      expect(stubNode.title).toBe('foo');
-
-      DOMPropertyOperations.deleteValueForProperty(stubNode, 'title');
-      expect(stubNode.getAttribute('title')).toBe(null);
-      // JSDOM does not behave correctly for attributes/properties
-      //expect(stubNode.title).toBe('');
+      var container = document.createElement('div');
+      ReactDOM.render(<div title="foo" />, container);
+      expect(container.firstChild.getAttribute('title')).toBe('foo');
+      expect(container.firstChild.title).toBe('foo');
+      ReactDOM.render(<div />, container);
+      expect(container.firstChild.getAttribute('title')).toBe(null);
+      expect(container.firstChild.title).toBe('');
     });
 
     it('should not remove attributes for special properties', () => {
-      stubNode = document.createElement('input');
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-
-      stubNode.setAttribute('value', 'foo');
-
-      DOMPropertyOperations.deleteValueForProperty(stubNode, 'value');
-      // JSDOM does not behave correctly for attributes/properties
-      //expect(stubNode.getAttribute('value')).toBe('foo');
-      expect(stubNode.value).toBe('');
-    });
-
-    it('should not leave all options selected when deleting multiple', () => {
-      stubNode = document.createElement('select');
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-
-      stubNode.multiple = true;
-      stubNode.appendChild(document.createElement('option'));
-      stubNode.appendChild(document.createElement('option'));
-      stubNode.options[0].selected = true;
-      stubNode.options[1].selected = true;
-
-      DOMPropertyOperations.deleteValueForProperty(stubNode, 'multiple');
-      expect(stubNode.getAttribute('multiple')).toBe(null);
-      expect(stubNode.multiple).toBe(false);
-
-      expect(stubNode.options[0].selected && stubNode.options[1].selected).toBe(
-        false,
+      var container = document.createElement('div');
+      spyOn(console, 'error');
+      ReactDOM.render(
+        <input type="text" value="foo" onChange={function() {}} />,
+        container,
+      );
+      expect(container.firstChild.getAttribute('value')).toBe('foo');
+      expect(container.firstChild.value).toBe('foo');
+      ReactDOM.render(
+        <input type="text" onChange={function() {}} />,
+        container,
+      );
+      expect(container.firstChild.getAttribute('value')).toBe('foo');
+      expect(container.firstChild.value).toBe('foo');
+      expect(console.error.calls.count()).toBe(1);
+      expect(console.error.calls.argsFor(0)[0]).toContain(
+        'A component is changing a controlled input of type text to be uncontrolled',
       );
```

Comment:
is this providing any additional coverage over the uncontrolled/controlled switch tests in `ReactDOMInput?